examples/torchscript/mnist_tensorboard_artifact.py

```diff
@@ -0,0 +1,180 @@
+
+    def log_weights(self, step):
+        writer.add_histogram('weights/conv1/weight', model.conv1.weight.data, step)
+        writer.add_histogram('weights/conv1/bias', model.conv1.bias.data, step)
+        writer.add_histogram('weights/conv2/weight', model.conv2.weight.data, step)
+        writer.add_histogram('weights/conv2/bias', model.conv2.bias.data, step)
+        writer.add_histogram('weights/fc1/weight', model.fc1.weight.data, step)
+        writer.add_histogram('weights/fc1/bias', model.fc1.bias.data, step)
+        writer.add_histogram('weights/fc2/weight', model.fc2.weight.data, step)
+        writer.add_histogram('weights/fc2/bias', model.fc2.bias.data, step)
+
+
+model = Net()
+if args.cuda:
+    model.cuda()
+
+optimizer = optim.SGD(model.parameters(), lr=args.lr, momentum=args.momentum)
+
+writer = None # Will be used to write TensorBoard events
+
+
+def train(epoch):
+    model.train()
+    for batch_idx, (data, target) in enumerate(train_loader):
+        if args.cuda:
+            data, target = data.cuda(), target.cuda()
+        data, target = Variable(data), Variable(target)
+        optimizer.zero_grad()
+        output = model(data)
+        loss = F.nll_loss(output, target)
+        loss.backward()
+        optimizer.step()
+        if batch_idx % args.log_interval == 0:
+            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
+                epoch, batch_idx * len(data), len(train_loader.dataset),
+                100. * batch_idx / len(train_loader), loss.data.item()))
+            step = epoch * len(train_loader) + batch_idx
+            log_scalar('train_loss', loss.data.item(), step)
+            model.log_weights(step)
+
+
+def test(epoch):
+    model.eval()
+    test_loss = 0
+    correct = 0
+    with torch.no_grad():
+        for data, target in test_loader:
+            if args.cuda:
+                data, target = data.cuda(), target.cuda()
+            data, target = Variable(data), Variable(target)
+            output = model(data)
+            test_loss += F.nll_loss(output, target, reduction='sum').data.item() # sum up batch loss
+            pred = output.data.max(1)[1] # get the index of the max log-probability
+            correct += pred.eq(target.data).cpu().sum().item()
+
+    test_loss /= len(test_loader.dataset)
+    test_accuracy = 100.0 * correct / len(test_loader.dataset)
+    print('\nTest set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n'.format(
+        test_loss, correct, len(test_loader.dataset), test_accuracy))
+    step = (epoch + 1) * len(train_loader)
+    log_scalar('test_loss', test_loss, step)
+    log_scalar('test_accuracy', test_accuracy, step)
+
+
+def log_scalar(name, value, step):
+    """Log a scalar value to both MLflow and TensorBoard"""
+    writer.add_scalar(name, value, step)
+    mlflow.log_metric(name, value)
+
+
+with mlflow.start_run() as run:
+    # Log our parameters into mlflow
+    for key, value in vars(args).items():
```

Comment:
Could we use `mlflow.log_params(var(args))` to log all these params in a single line / API request? Just to encourage best practices for efficient logging in our examples

Reply:
Sure, I have copied this script from [the pytorch example](https://github.com/mlflow/mlflow/blob/d8e6a8f07ce8fa940e26a68f0fe19f9fda1fb1da/examples/pytorch/mnist_tensorboard_artifact.py#L150). I am just wondering if the author had a reason to do it this way

Reply:
Got it sounds good! Yeah I actually wonder if we could just update the pytorch example to log torchscript models, vs introducing a new example snippet that's a copy of the pytorch snippet (because we do want to encourage pytorch users to log torchscript models)

Reply:
That's a fair reasoning. Let me know what's the final decision and I'll change it accordingly. In my opinion, you need to show both (you can put both in the same example) but then that would negate the structuring of the project, which is one folder for every flavour.